### ISSUE TITLE
pylint can't see parent methods through Generic inheritance

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#4080]
The following code:
```python3
from typing import cast, Generic, TypeVar

class A():
    data = 'OK'

    def getValue(self) -> str:
        return self.data

class B(A):
    pass
class C(B):
    pass

print('C:', C().getValue())

T = TypeVar('T')

class AG(Generic[T]):
    data = cast(T, 'OK')

    def getValue(self) -> T:
        return self.data

class BG(AG[T]):
    pass
class CG(BG[str]):
    pass

print('CG:', CG().getValue())
```
produces the following ouput:
```python3
$ python3 test.py
C: OK
CG: OK

$ mypy test.py
Success: no issues found in 1 source file

$ pylint test.py
************* Module test
test.py:29: [E1101(no-member), ] Instance of 'CG' has no 'getValue' member
------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 5.50/10, +2.00)

$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]
```
Note the `AG`-`BG`-`CG` class hierarchy is identical to `A`-`B`-`C` class hierarchy except it uses `Generic`s.
And `no-member` is reported in generic case while non-generic case is fine.

**Comment by hippo91** [comment]
@jolaf thanks for the report. I can reproduce it with this smaller snippet:
```
#pylint: disable=missing-module-docstring, invalid-name
#pylint: disable=missing-class-docstring, missing-function-docstring
#pylint: disable=too-few-public-methods
from typing import Generic, TypeVar

T = TypeVar('T')

class AG(Generic[T]):
    data = "my_data"

    def getValue(self) -> T:
        return self.data

class BG(AG[T]):
    pass
class CG(BG[str]):
    pass

print('BG:', BG().getValue())
print('CG:', CG().getValue())
```

**Comment by hippo91** [comment]
@jolaf i will close this issue as it seems to be a duplicate of #3131.
By the way this issue should be fixed as soon as PyCQA/astroid#927 will be merged into master.
